### PR TITLE
Allow some HTML markup in the description and full_article fields

### PR DESCRIPTION
**Pull request by mfairchild365**
HTML markup is limited in what elements and attributes can be used. It is also purified to prevent XSS attacks and fix invalid code.

description field allows
* a[href]
* strong
* p
* em

full_article field allows: 
* a[href]
* strong
* p
* em
* ul
* ol
* li

Lists are not allowed in the description field because they go against the purpose of a description (summary) paragraph and they don't play nice with images that are floated left/right in emails.

I'm not sure if I should provide a description of what elements are allowed for which fields in the interface. I almost don't want to encourage using these elements (because email clients are very picky), but maybe it wouldn't be a bad thing.

Demo is available here: 
http://ucommfairchild.unl.edu/mfairchild365/unl_enews/www/

### CODE REVIEW

**Comment by mfairchild365** [comment]
This needs some more testing before it gets merged.

File: www/js/submission.js [hunk 1]
```diff
@@ -276,9 +276,18 @@ var submission = function($) {
 		},
 
 		updatePreview : function() {
-			$('#sampleLayout p').text(function(index){
+			$('#sampleLayout p').html(function(index){
 				if ($('#description').val().length) {
-					return $('#description').val().substring(0,300);
+					var string = $('#description').val().substring(0,300);
+					console.log(string);
```
Comment (kabel) [inline www/js/submission.js]: Debugging line

File: www/js/submission.js [hunk 2]
```diff
@@ -276,9 +276,18 @@ var submission = function($) {
...
+					//Purify it to match what the actual output would look like
+					string = DOMPurify.sanitize(string, {
+						ALLOWED_TAGS: ['a','strong','p','em'],
```
Comment (kabel) [inline www/js/submission.js]: Probably best to hoist these configuration settings to the top of the file where they can be quickly changed.

File: www/js/submission.js [hunk 3]
```diff
@@ -276,9 +276,18 @@ var submission = function($) {
...
+						ALLOWED_ATTR: ['href'],
+						SAFE_FOR_JQUERY: true
+					});
+					console.log(string);
```
Comment (kabel) [inline www/js/submission.js]: Debugging statement

File: www/templates/html/ENews/Story/field-description.tpl.php [hunk 4]
```diff
@@ -0,0 +1,10 @@
+<?php
+$config = HTMLPurifier_Config::createDefault();
+$config->set('HTML.Allowed', "a[href],strong,p,em");
```
Comment (kabel) [inline www/templates/html/ENews/Story/field-description.tpl.php]: Same thing about centralizing config, if possible

File: www/templates/html/ENews/Submission.tpl.php [hunk 5]
```diff
@@ -42,6 +42,7 @@ function getValue($object, $field)
 </script>
 <script type="text/javascript" src="/wdn/templates_3.0/scripts/plugins/ui/jQuery.ui.js"></script>
 <script type="text/javascript" src="<?php echo UNL_ENews_Controller::getURL();?>js/submission.js"></script>
+<script type="text/javascript" src="<?php echo UNL_ENews_Controller::getURL();?>js/purify.js"></script>
```
Comment (kabel) [inline www/templates/html/ENews/Submission.tpl.php]: Let's set our goal for html5 style to not include the script types.

File: www/templates/html/ENews/Submission.tpl.php [hunk 6]
```diff
@@ -228,7 +229,7 @@ function getValue($object, $field)
         <?php }  ?>
     </div>
     <p>&lt;Enter Your Article Text&gt;</p>
-    <a href="#"></a>
+    <a href="#" class="supporting_website"></a>
```
Comment (kabel) [inline www/templates/html/ENews/Submission.tpl.php]: Maybe move this to JS rendering

Comment (mfairchild365) [inline www/templates/html/ENews/Submission.tpl.php]: Good idea

**Comment by ericras** [comment]
URL included in the Full Article is not converted to a link in the 'preview' area.

**Comment by mfairchild365** [comment]
@ericras UNLs included in the DESCRIPTION field were not being converted into links in the preview area, is that what you mean? The full article appears field to be working correctly.

**Pull request merged**